### PR TITLE
CR 1217061 : xrt-smi advanced --help prints Alveo help

### PR DESCRIPTION
**Pull request by aktondak**
#### Problem solved by the commit
xrt-smi advanced --help prints Alveo help which is incorrect

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
https://jira.xilinx.com/browse/CR-1217061
Discovered through internal testing

#### How problem was solved, alternative solutions (if any) and why they were rejected
The problem was solved by setting m_device member and using to print specific help. 
--report had multiple other bugs which are resolved with this :
1. mode was used as an option which showed --mode as a user configurable option in help. This was incorrect.
2. "--report clocks" was not being passed to sub-option making the parser show undefined behavior. 
3. There were no prints of what reports can be generated. Added "clocks" and "preemption" as valid reports.

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
Tested on kracken board :
Before fix
```
Y:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --help

COMMAND: advanced

DESCRIPTION: Low level command operations.

USAGE: xrt-smi advanced [ --read-mem | --write-mem | --report ] [--help]

OPTIONS:
 Common:
  --help             - Help to use this sub-command
 AIE:
  --report           - Hidden reports
 Alveo:
  --read-mem         - Read from the given memory address
  --write-mem        - Write to a given memory address


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
```
After fix
```
Y:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --help

COMMAND: advanced

DESCRIPTION: Low level command operations.

USAGE: xrt-smi advanced [ --report ] [--help] [-d arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  --help             - Help to use this sub-command
  --report           - Hidden reports


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
```
#### Documentation impact (if any)
None

### CODE REVIEW

**Review by AShivangi — COMMENTED** [review]
You tried out all possible combinations for this? Passing in invalid values as well?

**Comment by aktondak** [comment]
> You tried out all possible combinations for this? Passing in invalid values as well?

Passing an invalid value yields unrecognized arguments error :
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --report clock
ERROR: Unrecognized arguments:
  clock


COMMAND: advanced

DESCRIPTION: Low level command operations.

USAGE: xrt-smi advanced [ --read-mem | --write-mem | --report ] [--help] [-d arg]

OPTIONS:
 Common:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  --help             - Help to use this sub-command
 AIE:
  --report           - Reports to generate: clocks, preemption
 Alveo:
  --read-mem         - Read from the given memory address
  --write-mem        - Write to a given memory address


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
ERROR: operation canceled
```

File: src/runtime_src/core/tools/xbutil2/SubCmdAdvanced.cpp [hunk 1]
```diff
@@ -46,6 +46,7 @@ SubCmdAdvanced::SubCmdAdvanced(bool _isHidden, bool _isDepricated, bool _isPreli
   setIsPreliminary(_isPreliminary);
 
   m_commonOptions.add_options()
+    ("device,d", boost::program_options::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdAdvanced.cpp:49]: Do we need this device. "xrt-smi advanced --device <>" Is not a valid command. We have devices inside the suboptions

Comment (AShivangi) [reply src/runtime_src/core/tools/xbutil2/SubCmdAdvanced.cpp:49]: Ok, please make sure that all possible option combos work after this change!

Comment (aktondak) [reply src/runtime_src/core/tools/xbutil2/SubCmdAdvanced.cpp:49]: Only clocks and preemption are used as valida reports right now. Generating preemption exits silently and clocks works fine :
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --report preemption

Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --report clocks

Clocks
  H Clock                : 800 MHz
  MP-NPU Clock           : 400 MHz
```

**Comment by AShivangi** [comment]
> > You tried out all possible combinations for this? Passing in invalid values as well?
> 
> Passing an invalid value yields unrecognized arguments error :
> 
> ```
> Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --report clock
> ERROR: Unrecognized arguments:
>   clock
> 
> 
> COMMAND: advanced
> 
> DESCRIPTION: Low level command operations.
> 
> USAGE: xrt-smi advanced [ --read-mem | --write-mem | --report ] [--help] [-d arg]
> 
> OPTIONS:
>  Common:
>   -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
>   --help             - Help to use this sub-command
>  AIE:
>   --report           - Reports to generate: clocks, preemption
>  Alveo:
>   --read-mem         - Read from the given memory address
>   --write-mem        - Write to a given memory address
> 
> 
> GLOBAL OPTIONS:
>   --verbose          - Turn on verbosity
>   --batch            - Enable batch mode (disables escape characters)
>   --force            - When possible, force an operation
> ERROR: operation canceled
> ```

Why is unrecognized option showing both aie and alveo?

**Comment by aktondak** [comment]
> > > You tried out all possible combinations for this? Passing in invalid values as well?
> > 
> > 
> > Passing an invalid value yields unrecognized arguments error :
> > ```
> > Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --report clock
> > ERROR: Unrecognized arguments:
> >   clock
> > 
> > 
> > COMMAND: advanced
> > 
> > DESCRIPTION: Low level command operations.
> > 
> > USAGE: xrt-smi advanced [ --read-mem | --write-mem | --report ] [--help] [-d arg]
> > 
> > OPTIONS:
> >  Common:
> >   -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
> >   --help             - Help to use this sub-command
> >  AIE:
> >   --report           - Reports to generate: clocks, preemption
> >  Alveo:
> >   --read-mem         - Read from the given memory address
> >   --write-mem        - Write to a given memory address
> > 
> > 
> > GLOBAL OPTIONS:
> >   --verbose          - Turn on verbosity
> >   --batch            - Enable batch mode (disables escape characters)
> >   --force            - When possible, force an operation
> > ERROR: operation canceled
> > ```
> 
> Why is unrecognized option showing both aie and alveo?

Apologies, this is outdated.
The updated behavior prints Ryzen help only :
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi advanced --report clock

ERROR: Invalid report value: 'clock'
: invalid argument

COMMAND: advanced

DESCRIPTION: Reports to generate: clocks, preemption

USAGE:  advanced --report [--help] [-d arg]

OPTIONS:
  --report           - Reports to generate: clocks, preemption
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  --help             - Help to use this sub-command


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
```

**Review by rchane — APPROVED** [review]
Output seems good after changes!

**Review by AShivangi — APPROVED** [review]

**Pull request merged**